pkg/oauth/apis/oauth/validation/validation_test.go

```diff
@@ -746,3 +746,38 @@ func TestValidateAuthorizeTokensUpdate(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateUserNameField(t *testing.T) {
+	fieldArg := field.NewPath("metadata", "name")
+	valid := "kube:admin"
+	errs := ValidateUserNameField(valid, fieldArg)
+	if len(errs) != 0 {
+		t.Errorf("expected success: %v", errs)
+	}
+
+	errorCases := map[string]struct {
+		name             string
+		expectedErrType  field.ErrorType
+		expectedErrField string
+	}{
+		"system:admin": {
+			expectedErrType:  field.ErrorTypeInvalid,
+			expectedErrField: "metadata.name",
+		},
+	}
+	for name, tc := range errorCases {
+		errs := ValidateUserNameField(name, fieldArg)
+		if len(errs) == 0 {
+			t.Errorf("%s: expected 1 error got 0", name)
+			continue
+		}
```

Comment:
This is more of a legacy way of how unit tests used to be written (and I see it all over this repo, so I understand).

The more current way is more similar to what you can see here:
https://github.com/enj/cluster-authentication-operator/blob/1ca2f4502798fe963b76584678d8fc341d91b2b2/pkg/controllers/configobservation/oauth/observe_templates_test.go#L19

Notice the
```golang
tests := []struct {
		name           
```
- so no map, you already have the name be the part of the struct + you are able to determine the order in which the tests will run

and
```golang
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			indexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
```
which I believe is something @s-urbaniak might have mentioned already.

This is the tabular style of tests that comes from https://github.com/cweill/gotests (which allows allows autogeneration of such test structures in VS Code).

Reply:
I regenerated the tests using the latest test templates from Goland. Should be fine now.

As a side note - is there any Goland formatter with test templates? I'm asking because without it, we'll always struggle to death with what convention to use when writing tests, sorting imports etc. It's spaces vs tabs or emacs vs vi wars ;)